gemrb/plugins/GUIScript/GUIScript.cpp

```diff
@@ -2564,6 +2564,48 @@ static PyObject* GemRB_Control_SetEvent(PyObject * /*self*/, PyObject* args)
 	Py_RETURN_NONE;
 }
 
+PyDoc_STRVAR( GemRB_Window_SetKeyPressEvent__doc,
+"===== Window_SetKeyPressEvent =====\n\
+\n\
+**Prototype:** _GemRB.Window_SetKeyPressEvent (callback)\n\
+\n\
+**Description:** Sets a callback function to handle key press event on window scopes. \n\
+\n\
+**Parameters:**\n\
+  * callback - Python function that accepts (windowIndex, key, mod) arguments and returns\n\
+	* 					 1 indicating succesful key press consumption or 0 otherwise. \n\
+\n\
+**Return value:** N/A\n\
+\n\
+**Example:**\n\
+    Window.SetKeyPressEvent (KeyPressCallback)\n\
+    return\n\
+\n\
+**See also:** [[guiscript:QuitGame]]\n\
+"
+);
+
+static PyObject* GemRB_Window_SetKeyPressEvent(PyObject* /*self*/, PyObject *args) {
+	int windowIndex;
+	PyObject *fn;
+
+	if(!PyArg_ParseTuple(args, "iO", &windowIndex, &fn)) {
+		return AttributeError(GemRB_Window_SetKeyPressEvent__doc);
+	}
+
+	WindowKeyPressHandler handler = NULL;
+	if(fn != Py_None && PyCallable_Check(fn)) {
```

Comment:
doesn't this make it impossible to turn off a callback without creating a noop function first?